helix-loader/src/grammar.rs

```diff
@@ -311,13 +320,14 @@ fn build_tree_sitter_library(src_path: &Path, grammar: GrammarConfiguration) ->
         .opt_level(3)
         .cargo_metadata(false)
         .host(BUILD_TARGET)
-        .target(BUILD_TARGET);
+        .target(target.unwrap_or(BUILD_TARGET));
```

Comment:
When would `target` be different from `BUILD_TARGET`? Both come from `TARGET` environment variables in build scripts, `BUILD_TARGET` from the `helix-loader` crate's build script and `target` from `helix-term`'s

Reply:
When `helix_loader::build_grammars` is being used for cross compilation.

`BUILD_TARGET` represents the target on which the code of `helix-loader` runs. `helix-term`'s build script (which has `helix-loader` as build-dependency) needs to use `helix-loader` to cross-compile for `helix-term`'s target, which may be different from `BUILD_TARGET`.

E.g., when cross-compiling `helix-term` to an `aarch64` target on an `x86_64` host, two binaries of `helix-loader` are built:

- One binary for `aarch64` target for use by `helix-term`'s main crate (normal dependency)
    - `helix-loader`'s build script runs with `TARGET = "aarch64"`
    - Hence, `env!("BUILD_TARGET") = "aarch64"`
    - `helix-term`'s main crate code passes `target = None`  because it wants to compile grammars for the current host = `helix-loader`'s build target

- One binary for `x86_64` target for use by `helix-term`'s build script (build dependency)
    - `helix-loader`'s build script runs with `TARGET = "x86_64"`
    - Hence, `env!("BUILD_TARGET") = "x86_64"`
    - `helix-term`'s build script passes `target = "aarch64"`  because it wants to cross-compile grammars for `helix-term`'s build target, not the host on which the build script runs